### PR TITLE
feat: implement just in time funding feature

### PR DESCRIPTION
**Pull request by luzzif**

### CODE REVIEW

File: packages/contracts/test/ERC20KPITokenCollectProtocoFeeTest.sol [hunk 1]
```diff
@@ -39,32 +44,103 @@ contract ERC20KPITokenCollectProtocoFeeTest is BaseTestSetup {
...
+                creator: address(this),
+                oraclesManager: address(oraclesManager),
+                kpiTokensManager: address(kpiTokensManager),
+                feeReceiver: feeReceiver,
+                kpiTokenTemplateId: 1,
+                kpiTokenTemplateVersion: 1,
+                description: "a",
+                expiration: block.timestamp + 60,
+                kpiTokenData: abi.encode(rewards, "Token", "TKN", 100 ether, false),
+                oraclesData: abi.encode(oracleData, false)
```
Comment (guerrap) [inline packages/contracts/test/ERC20KPITokenCollectProtocoFeeTest.sol:105]: Aren't we missing the `_jitFunding` param here?

File: packages/contracts/test/ERC20KPITokenInitializeTest.sol [hunk 2]
```diff
@@ -28,7 +29,7 @@ contract ERC20KPITokenInitializeTest is BaseTestSetup {
                 kpiTokenTemplateVersion: 1,
                 description: "a",
                 expiration: block.timestamp + 60,
-                kpiTokenData: abi.encode(collaterals, "Token", "TKN", 100 ether),
+                kpiTokenData: abi.encode(rewards, "Token", "TKN", 100 ether),
```
Comment (guerrap) [inline packages/contracts/test/ERC20KPITokenInitializeTest.sol]: Aren't we missing the `_jitFunding` here? In the `abi.encode()`.

File: packages/contracts/test/ERC20KPITokenInitializeTest.sol [hunk 3]
```diff
@@ -153,7 +154,7 @@ contract ERC20KPITokenInitializeTest is BaseTestSetup {
                 kpiTokenTemplateVersion: 1,
                 description: "a",
                 expiration: block.timestamp + 60,
-                kpiTokenData: abi.encode(collaterals, "Token", "TKN", 10 ether),
+                kpiTokenData: abi.encode(rewards, "Token", "TKN", 10 ether),
```
Comment (guerrap) [inline packages/contracts/test/ERC20KPITokenInitializeTest.sol]: Same as above.

Comment (luzzif) [reply packages/contracts/test/ERC20KPITokenInitializeTest.sol]: Should be addressed in b899106e6cd269c8a3110a534a7064440c865e89

File: packages/contracts/test/ERC20KPITokenInitializeTest.sol (hunk 2 above)
Comment (luzzif) [reply packages/contracts/test/ERC20KPITokenInitializeTest.sol]: Should be addressed in b899106e6cd269c8a3110a534a7064440c865e89

**Pull request merged**